### ISSUE TITLE
Importance of F.relu()

### ISSUE THREAD

**Issue by MichaelLiang12** [adambielski/siamese-triplet#5]
In your definition, there is no `max(0,x)` used.[https://github.com/adambielski/siamese-triplet#triplet-network](https://github.com/adambielski/siamese-triplet#triplet-network)

But in your code, you use a `F.relu()` to get similar effects with `max(0,x)`. Is there crucial differences between these two definitions with or without `F.relu()`? What about their characteristics and performance?

**Comment by adambielski** [comment]
There's no difference, even Relu implementation in PyTorch is a wrapper of a more general `threshold` function with threshold and replacement values set to 0. Alternatively `clamp` would do the trick.

**Comment by MichaelLiang12** [comment]
Sorrry I made myself misunderstood. I mean, in your instruction, there is no `max(0,x)` used. 
![triplet_loss](https://user-images.githubusercontent.com/17873150/37882486-415d58a4-30d6-11e8-81c4-8d096b4c3325.png)
But in your codes, you use the `max(0,x)` term.
`losses = F.relu(distance_positive - distance_negative + self.margin)`

 Any difference between them? What about their characteristics and performance?

**Comment by adambielski** [comment]
Thanks for catching this, I fixed it in readme. The margin value has no use without the max - it disappears when computing the gradient. Also, the function would not have a minimum value so it's hard to optimize
